### PR TITLE
Ad hoc filtering of stack frames to improve debugging experience

### PR DESCRIPTION
**Pull request by erights**
First, let me acknowledge that this PR adds a configuration option in an ad-hoc and non-general way. The proper way would be for the filtering criteria to be more generally customizable by data, rather than building a particular filter in. This would require us to figure out how to pass in complex customization. In the meantime, for our current development practices, these simple ad-hoc filters seem to work rather well.

This PR adds a new `errorTaming` option `'unfiltered'`. The behavior with this option preserves the `'safe'` behavior before this PR. The behavior of the `'safe'` and `'unsafe'` options ***on v8*** now filters stack frames in terms of the ad-hoc rules added by this PR.

Outside of v8, this PR should have no effect. Whether filtered on unfiltered, either one is consistent with the current state of https://tc39.es/proposal-error-stacks/#sec-system since it leaves the contents of the error stack up to the host.

### CODE REVIEW

**Comment by erights** [comment]
To illustrate. Running `yarn qt test/error/tame-console.test.js` from inside `packages/ses` produces:

```
$ yarn ava test/error/tame-console.test.js
$ /Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/.bin/ava test/error/tame-console.test.js

bar happens (Error#1)

  Error#1: caused by (SyntaxError#2) , Object <[Object: null prototype] {}> {}
   
    at ../../packages/ses/test/error/tame-console.test.js:49:12
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Promise.all (index 1)
  
  Error#1
    SyntaxError#2: foo
     
      at ../../packages/ses/test/error/tame-console.test.js:48:20
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at async Promise.all (index 1)
    
bar happens (URIError#3)

  URIError#3: bar
   
    at ../../packages/ses/test/error/tame-console.test.js:78:18
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Promise.all (index 3)
  
  URIError#3 ERROR_NOTE: caused by (SyntaxError#4) , Object <[Object: null prototype] {}> {}
  URIError#3
    SyntaxError#4: foo
     
      at ../../packages/ses/test/error/tame-console.test.js:77:18
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at async Promise.all (index 3)
```

whereas running `yarn qt test/error/tame-console-unfilteredError.test.js` from inside `packages/ses` produces:

```
$ yarn ava test/error/tame-console-unfilteredError.test.js
$ /Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/.bin/ava test/error/tame-console-unfilteredError.test.js

bar happens (Error#1)

  Error#1: caused by (SyntaxError#2) , Object <[Object: null prototype] {}> {}
   
    at makeError (file:///Users/markmiller/src/ongithub/agoric/SES-shim/packages/ses/src/error/assert.js:166:17)
    at Function.fail (file:///Users/markmiller/src/ongithub/agoric/SES-shim/packages/ses/src/error/assert.js:305:19)
    at file:///Users/markmiller/src/ongithub/agoric/SES-shim/packages/ses/test/error/tame-console-unfilteredError.test.js:41:12
    at Test.callFn (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/test.js:611:21)
    at Test.run (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/test.js:624:23)
    at Runner.runSingle (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:281:33)
    at Runner.runTest (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:349:30)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Promise.all (index 1)
    at async /Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:496:21
  
  Error#1
    SyntaxError#2: foo
     
      at file:///Users/markmiller/src/ongithub/agoric/SES-shim/packages/ses/test/error/tame-console-unfilteredError.test.js:40:20
      at Test.callFn (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/test.js:611:21)
      at Test.run (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/test.js:624:23)
      at Runner.runSingle (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:281:33)
      at Runner.runTest (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:349:30)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at async Promise.all (index 1)
      at async /Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:496:21
      at async Runner.start (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:506:15)
    
bar happens (URIError#3)

  URIError#3: bar
   
    at file:///Users/markmiller/src/ongithub/agoric/SES-shim/packages/ses/test/error/tame-console-unfilteredError.test.js:59:18
    at Test.callFn (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/test.js:611:21)
    at Test.run (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/test.js:624:23)
    at Runner.runSingle (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:281:33)
    at Runner.runTest (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:349:30)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Promise.all (index 3)
    at async /Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:496:21
    at async Runner.start (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:506:15)
  
  URIError#3 ERROR_NOTE: caused by (SyntaxError#4) , Object <[Object: null prototype] {}> {}
  URIError#3
    SyntaxError#4: foo
     
      at file:///Users/markmiller/src/ongithub/agoric/SES-shim/packages/ses/test/error/tame-console-unfilteredError.test.js:58:18
      at Test.callFn (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/test.js:611:21)
      at Test.run (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/test.js:624:23)
      at Runner.runSingle (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:281:33)
      at Runner.runTest (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:349:30)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at async Promise.all (index 3)
      at async /Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:496:21
      at async Runner.start (/Users/markmiller/src/ongithub/agoric/SES-shim/node_modules/ava/lib/runner.js:506:15)
```

**Comment by erights** [comment]
This is now ready for review.

**Comment by erights** [comment]
Still not yet ready for review.

**Comment by erights** [comment]
Hi @michaelfig @dckc this is finally ready for review again. PTAL.

File: packages/ses/src/error/tame-v8-error-constructor.js [hunk 1]
```diff
@@ -39,20 +39,45 @@ const safeV8CallSiteFacet = callSite => {
...
+      )
+    );
+  };
+
+  const callSiteStringifier = callSite => {
+    let callSiteString = `${callSite}`;
+    if (stackFiltering === 'concise') {
+      const match = FILENAME_FILTER.exec(callSiteString);
+      if (match) {
+        callSiteString = `${match[1]}${match[2]}`;
```
Comment (michaelfig) [inline packages/ses/src/error/tame-v8-error-constructor.js:72]: It seems to me that `match[2]` might be `undefined`.  We wouldn't want that to leak into the resulting string.  Please write a test case for that.

```suggestion
        callSiteString = `${match[1]}${match[2] || ''}`;
```

**Review by michaelfig — APPROVED** [review]
Getting close.  I just need assurance about the regexp issue I pointed out.

File: packages/ses/src/error/tame-v8-error-constructor.js (hunk 1 above)
Comment (erights) [reply packages/ses/src/error/tame-v8-error-constructor.js:72]: Hi @michaelfig I'm happy to write a test case, but I don't understand. The pattern definition is
```js
const FILENAME_FILTER = /^((?:.*[( ])?)[:/\w-_]*\/(packages\/.+)$/;
```
I cannot see how a match could be successful without both `match[1]` and `match[2]` being strings. Can you think of a test case string for the pattern that could produce the outcome you're worried about?

Comment (erights) [reply packages/ses/src/error/tame-v8-error-constructor.js:83]: I added a comment above `FILENAME_FILTER` to clarify.

I also found out how to do collapsible sections in markdown, and used it to add some example log output to the `lockdown-options.md` page. Collapsible sections are cool!

Comment (michaelfig) [reply packages/ses/src/error/tame-v8-error-constructor.js:83]: Okay, that was my misreading.  Testing it shows it doesn't have the problem I was worried about.

**Review by michaelfig — APPROVED** [review]
LGTM!

**Pull request merged**